msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthHttpRequest.java

```diff
@@ -55,39 +54,31 @@ private Map<String, String> configureHttpHeaders() {
         Map<String, String> httpHeaders = new HashMap<>(extraHeaderParams);
         httpHeaders.put("Content-Type", HTTPContentType.ApplicationURLEncoded.contentType);
 
-        if (this.getAuthorization() != null) {
-            httpHeaders.put("Authorization", this.getAuthorization());
-        }
-
         Map<String, String> telemetryHeaders =
                 serviceBundle.getServerSideTelemetry().getServerTelemetryHeaderMap();
         httpHeaders.putAll(telemetryHeaders);
 
         return httpHeaders;
     }
 
-    private HTTPResponse createOauthHttpResponseFromHttpResponse(IHttpResponse httpResponse)
+    private HttpResponse createOauthHttpResponseFromHttpResponse(IHttpResponse httpResponse)
             throws IOException {
 
-        final HTTPResponse response = new HTTPResponse(httpResponse.statusCode());
+        final HttpResponse response = new HttpResponse();
+        response.statusCode(httpResponse.statusCode());
 
         final String location = HttpUtils.headerValue(httpResponse.headers(), "Location");
         if (!StringHelper.isBlank(location)) {
             try {
-                response.setLocation(new URI(location));
+                response.addHeader("Location", new URI(location).toString());
```

Comment:
Maybe worth it to have an enum/constants for well known headers.

Reply:
I was planning to do this in a follow up PR that removes various nimbus utility methods, since we'll then have the full picture of what constants and enums would be useful